core/imagedownloader.cpp

```diff
@@ -65,45 +65,57 @@ void ImageDownloader::saveImage(QNetworkReply *reply)
 	reply->deleteLater();
 }
 
-static void loadPicture(QUrl url, QString filename)
+// Fetch a picture from the given filename. If this is a non-remote filename, fetch it from disk.
+// Remote files are fetched from the net in a background thread. In such a case, the output-flag
+// "stillLoading" is set to true.
+// If the input-flag "tryDownload" is set to false, no download attempt is made. This is to
+// prevent infinite loops, where failed image downloads would be repeated ad infinitum.
+// Returns: fetched image, stillLoading flag
+static std::pair<QImage,bool> fetchImage(const QString &filename, const QString &originalFilename, bool tryDownload)
```

Comment:
a small nit would be to always add spaces between the template types:
`<QImage, bool>`

i'm ok with pair, tuple, tie and so on, and while these are new and cool (although, i just found a blog post that explains how these "smell"), i wonder how the rest of our developers feel about these.

Reply:
> a small nit would be to always add spaces between the template types:

done.

> i'm ok with pair, tuple, tie and so on, and while these are new and cool (although, i just found a blog post that explains how these "smell"), i wonder how the rest of our developers feel about these.

I think that once it's more complex than a pair, a struct is probably more appropriate. The one occcasion where `tie` is really nice is for comparisons. Note that it's all in C++11, so not all that new. I can replace by a struct, but would prefer doing this with the video branch, when all this is more settled. Notably, in the video branch the `stillLoading` bool is replaced by a `mediatype_t` which can adopt quite a number of different states (picture, video, io-error, still-loading, unknown).